src/liballoc/boxed.rs

```diff
@@ -249,6 +249,14 @@ impl<T: ?Sized> Box<T> {
     /// This function is unsafe because improper use may lead to
     /// memory problems. For example, a double-free may occur if the
     /// function is called twice on the same raw pointer.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// let mut x = 5;
+    /// let ptr = &mut x as *mut i32;
+    /// let b = unsafe { Box::from_raw(ptr) };
```

Comment:
The Box ends up being a pointer to the stack, and is then freed when the
Box is dropped.

On Saturday, July 9, 2016, Guillaume Gomez <notifications@github.com> wrote:

> In src/liballoc/boxed.rs
> <https://github.com/rust-lang/rust/pull/34740#discussion_r70173240>:
>
> > @@ -249,6 +249,14 @@ impl<T: ?Sized> Box<T> {
> >      /// This function is unsafe because improper use may lead to
> >      /// memory problems. For example, a double-free may occur if the
> >      /// function is called twice on the same raw pointer.
> > +    ///
> > +    /// # Examples
> > +    ///
> > +    /// ```
> > +    /// let mut x = 5;
> > +    /// let ptr = &mut x as *mut i32;
> > +    /// let b = unsafe { Box::from_raw(ptr) };
>
> I see. Just by curiosity: why would it likely cause a segfault in here?
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/rust-lang/rust/pull/34740/files/297695d76da3a624e380ad9a190b8d3354605ef5#r70173240>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AIPosBuNJtBUz6ID429xaGrWDWV8j17rks5qUCdggaJpZM4JIqu_>
> .
>